ts/features/bonus/cgn/screens/CgnDetailScreen.tsx

```diff
@@ -52,22 +63,84 @@ import { eycaDetailSelector } from "../store/reducers/eyca/details";
 import { cgnUnsubscribeSelector } from "../store/reducers/unsubscribe";
 import { canEycaCardBeShown } from "../utils/eyca";
 import { availableBonusTypesSelectorFromId } from "../../common/store/selectors";
+import { useHeaderSecondLevel } from "../../../../hooks/useHeaderSecondLevel";
+import { openWebUrl } from "../../../../utils/url";
+import { EYCA_WEBSITE_DISCOUNTS_PAGE_URL } from "../utils/constants";
+import { CardRevoked } from "../../../../../definitions/cgn/CardRevoked";
+import { CardExpired } from "../../../../../definitions/cgn/CardExpired";
+import { formatDateAsShortFormat } from "../../../../utils/dates";
 
 type Props = ReturnType<typeof mapStateToProps> &
   ReturnType<typeof mapDispatchToProps>;
 
-const [gradFirstColor, gradSecondColor] = IOColorGradients.cgn;
+const HEADER_BACKGROUND_COLOR = "#8F84B7";
 
-const HEADER_BACKGROUND_COLOR = gradFirstColor;
-const GRADIENT_END_COLOR = gradSecondColor;
+const gradientSafeArea: IOSpacingScale = 80;
+const contentEndMargin: IOSpacingScale = 32;
+const spaceBetweenActions: IOSpacer = 24;
 
 /**
  * Screen to display all the information about the active CGN
  */
 const CgnDetailScreen = (props: Props): React.ReactElement => {
+  const headerHeight = useHeaderHeight();
   const [cardLoading, setCardLoading] = useState(true);
   const navigation =
     useNavigation<IOStackNavigationProp<CgnDetailsParamsList, "CGN_DETAILS">>();
+  const safeAreaInsets = useSafeAreaInsets();
+
+  const gradientOpacity = useSharedValue(1);
+  const scrollTranslationY = useSharedValue(0);
+
+  const bottomMargin: number = React.useMemo(
+    () =>
+      safeAreaInsets.bottom === 0
+        ? IOVisualCostants.appMarginDefault
+        : safeAreaInsets.bottom,
+    [safeAreaInsets]
+  );
+
+  const safeBottomAreaHeight: number = React.useMemo(
+    () => bottomMargin + buttonSolidHeight + contentEndMargin,
+    [bottomMargin]
+  );
+
+  const gradientAreaHeight: number = React.useMemo(
+    () => bottomMargin + buttonSolidHeight + gradientSafeArea,
+    [bottomMargin]
+  );
+
+  const footerGradientOpacityTransition = useAnimatedStyle(() => ({
+    opacity: withTiming(gradientOpacity.value, {
+      duration: 200,
+      easing: Easing.ease
+    })
+  }));
+
+  const scrollHandler = useAnimatedScrollHandler(
+    ({ contentOffset, layoutMeasurement, contentSize }) => {
+      // eslint-disable-next-line functional/immutable-data
+      scrollTranslationY.value = contentOffset.y;
+
+      const isEndReached =
+        Math.floor(layoutMeasurement.height + contentOffset.y) >=
+        Math.floor(contentSize.height);
+
+      // eslint-disable-next-line functional/immutable-data
+      gradientOpacity.value = isEndReached ? 0 : 1;
+    }
+  );
+
+  useHeaderSecondLevel({
+    title: I18n.t("bonus.cgn.name"),
+    goBack: navigation.goBack,
```

Comment:
```suggestion
```

Reply:
replaced it with just the attribute `canGoBack`

Reply:
commited in [78131ec](https://github.com/pagopa/io-app/pull/5651/commits/78131ec0640e722e829557e501c1805f234ea670)